docs/learn/documentation/versioned/jobs/configuration-table.html

```diff
@@ -822,6 +828,40 @@
                 </tr>
 
                 <tr>
+                    <th colspan="3" class="section" id="streams"><a href="../container/streams.html">Streams</a></th>
+                </tr>
+
+                <tr>
+                    <td class="property" id="streams-system">streams.<span class="stream">stream-id</span>.samza.system</td>
+                    <td class="default"></td>
+                    <td class="description">
+                        The <span class="system">system-name</span> of the system on which this stream will be accessed.
+                        This property binds the stream to one of the systems defined with the property
+                        systems.<span class="system">system-name</span>.samza.factory. <br>
+                        If this property isn't specified, it is inherited from job.default.system.
+                    </td>
+                </tr>
+
+                <tr>
+                    <td class="property" id="streams-physical-name">streams.<span class="stream">stream-id</span>.samza.physical.name</td>
```

Comment:
> The idea of introducing the logic stream id helps to:
> make the stream ID as the top-level configure variable and allow user program directly refer > to it 
> avoid fixation of physical stream name to the configure variable name
> avoid fixation of a stream's configuration to a system's configuration variable name

+1 for all the above. I am still not convinced with your user example because it is not the user code that should be dealing with the input parsing. But the ideas behind logical stream-id sound good to me. 
I am all for decoupling systems and streams configurations. We just need to make sure we roll it out correctly instead of removing the configs in one shot. 

> We got a chance to move forward w/ our API in this round. I view configuration also as part of the API and I would rather to clean it now than keeping both s.t. the configuration for systems and streams are muddy. We can still support the old deprecated configuration for now, but need to mark those deprecated configurations clear in the web doc.

Is there a way to do multiple +1 on this comment 👍  

> @jmakes the "re-write" I referred to in my comment was not referring to a config re-writer. 

Thanks for mentioning this Yi. I was going to oppose Jake's suggestion on a "rewriter" in the other comment :) I am fully opposed to any form of "config rewriter". It was a mistake we made in the first place and I would like to undo it at an opportune moment 👍